Source/WebCore/dom/ElementInternals.cpp

```diff
@@ -45,47 +46,21 @@ ShadowRoot* ElementInternals::shadowRoot() const
     return shadowRoot;
 }
 
-void ElementInternals::setRole(const AtomString& role)
+void ElementInternals::setAttributeWithoutSynchronization(const QualifiedName& name, const AtomString& value)
 {
-    setAriaValueForAttribute(roleAttr->localName(), role);
-}
-
-const AtomString& ElementInternals::role()
-{
-    return ariaValueForAttribute(roleAttr->localName());
-}
-
-void ElementInternals::setAriaRoleDescription(const AtomString& description)
-{
-    setAriaValueForAttribute(aria_roledescriptionAttr->localName(), description);
-}
-
-const AtomString& ElementInternals::ariaRoleDescription()
-{
-    return ariaValueForAttribute(aria_roledescriptionAttr->localName());
-}
-
-void ElementInternals::setAriaLabel(const AtomString& label)
-{
-    setAriaValueForAttribute(aria_labelAttr->localName(), label);
-}
-
-const AtomString& ElementInternals::ariaLabel()
-{
-    return ariaValueForAttribute(aria_labelAttr->localName());
-}
+    RefPtr element = m_element.get();
+    element->customElementDefaultARIA().setValueForAttribute(name, value);
 
-void ElementInternals::setAriaValueForAttribute(const AtomString& key, const AtomString& value)
-{
-    m_element->customElementDefaultARIA().setValueForAttribute(key, value);
+    if (AXObjectCache* cache = element->document().existingAXObjectCache())
+        cache->deferAttributeChangeIfNeeded(name, element.get());
```

Comment:
Unfortunately, a commit landed a few days ago that will require a change on this line.

https://github.com/WebKit/WebKit/commit/a313324db71fdc82a9ecfd4071e1569b1029281a

The signature of this function is now `AXObjectCache::deferAttributeChangeIfNeeded(Element* element, const QualifiedName& attrName, const AtomString& oldValue, const AtomString& newValue)`

Reply:
Fixed.